adr/0010-scripts-actions.md

```diff
@@ -0,0 +1,43 @@
+# 10. Scripts -> Actions
+
+Date: 2023-01-18
+
+## Status
+
+Accepted
+
+## Context
+
+Originally, the `scripts` noun was added to components to allow us to move hard-coded init business logic out of the codebase and into the package system. At the time there was only a `before` and `after` section with simply an array entry per command. Later, `prepare` was added as a way to do some thing during `zarf package create`. As teams began to find new ways to use the capabilities, their limitations became more obvious.
+
+## Decision
+
+The `scripts` section of the `zarf.yaml` will be replaced with a new `actions` section. The `actions` section will be a map of action names to a list of commands to run. `actions` will contain action sets that map to the following lifecycle events:
+
+- `onCreate` - Runs during `zarf package create`
+- `onDeploy` - Runs during `zarf package deploy`
+- `onRemove` - Runs during `zarf package remove`
+
+Each of these action sets are optional and contain different hooks and events that can be used to customize the package lifecycle:
+
+- `defaults` - action configuration that will be applied to all actions in the set
+- `before` - array of actions that will run before the component is processed for `create`, `deploy`, or `remove`
```

Comment:
I think using the term `the component` is a little odd since the action is part of the component.  maybe a more global phrase like "the other parts of the component" or "non action components" or something to make it clear that its the same component the action is defined in.